packages/cli/internal/pkg/cli/spec/context.go

```diff
@@ -10,10 +10,17 @@ type Filesystem struct {
 	Configuration FSConfig `yaml:"configuration,omitempty"`
 }
 type Engine struct {
-	Type       string     `yaml:"type"`
-	Engine     string     `yaml:"engine"`
-	Filesystem Filesystem `yaml:"filesystem,omitempty"`
+	Type                 string              `yaml:"type"`
+	Engine               string              `yaml:"engine"`
+	ResourceRequirements ResourceRequirement `yaml:"resourceRequirements,omitempty"`
+	Filesystem           Filesystem          `yaml:"filesystem,omitempty"`
 }
+
+type ResourceRequirement struct {
+	VCpus          int `yaml:"vcpus,omitempty"`
+	MemoryLimitMiB int `yaml:"memoryLimit,omitempty"`
```

Comment:
I don't think it will work as you might expect. "omitempty" means not to include values that absent. In your case values **always** present. 

Here is the test: 
https://go.dev/play/p/WB4AzY1JhTT

Reply:
I would suggest a unit test for that.